### PR TITLE
Fix typecheck error on Listing/Mapping

### PR DESCRIPTION
**Pull request by bioball**
The addresses the following case:

```
local l1: Listing<String> = new { 1 }
l2: Listing<Int> = (l1) { 2 }
```

The member in `l1` should be checked against its owner; to check against `Listing<String>`.

### CODE REVIEW

File: pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java [hunk 1]
```diff
@@ -298,6 +298,9 @@ public static Object doReadMember(
           }
         }
       } else if (receiver instanceof VmListingOrMapping<?> vmListingOrMapping) {
+        if (owner != receiver && owner instanceof VmListingOrMapping<?> vmListingOrMappingOwner) {
+          ret = vmListingOrMappingOwner.typecastObjectMember(member, ret, callNode);
```
Comment (stackoverflow) [inline pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java:302]: Why the `ret =` if in the next line you reassign to `ret`?

**Review by stackoverflow — APPROVED** [review]
Looks good.

File: pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java [hunk 2]
```diff
@@ -298,6 +298,9 @@ public static Object doReadMember(
...
+        }
         ret = vmListingOrMapping.typecastObjectMember(member, ret, callNode);
       }
```
Comment (translatenix) [inline pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java:305]: If the following suggestion doesn't make sense, a test is missing. :-)

```suggestion
      } else if (owner instanceof VmListingOrMapping<?> vmListingOrMappingOwner) {
        ret = vmListingOrMappingOwner.typecastObjectMember(member, ret, callNode);
      }
```

File: pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java (hunk 1 above)
Comment (holzensp) [reply pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java:302]: `ret` is also an argument to the function call in the line below; it doubles the cast. That said; @translatenix's suggestion should be sufficient.

**Review by holzensp — APPROVED** [review]
Approving to unblock; but take the suggestion or add the test ;)

File: pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java (hunk 1 above)
Comment (translatenix) [reply pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java:302]: Tests pass even after removing the first `ret =`.

**Pull request merged**